kong/plugins/acl/handler.lua

```diff
@@ -30,9 +30,15 @@ function ACLHandler:access(conf)
   ACLHandler.super.access(self)
 
   local consumer_id
-  if ngx.ctx.authenticated_credential then
+  if ngx.ctx.authenticated_consumer then
+    consumer_id = ngx.ctx.authenticated_consumer.id
+  end
+
+  if not consumer_id and ngx.ctx.authenticated_credential then
     consumer_id = ngx.ctx.authenticated_credential.consumer_id
-  else
+  end
+
```

Comment:
What I mean here is that even if there is `ngx.ctx.authenticated_consumer` table or `ngx.ctx. authenticated_credential` table,  there is no gurantee that it has any properties. We could go on, and even argue that the current and this patched code will crash if `ngx.ctx.authenticated_consumer` or `ngx.ctx. authenticated_credential` are not `tables`.